code/modules/mob/living/carbon/human/death.dm

```diff
@@ -47,12 +47,18 @@
 		ticker.mode.update_cult_icons_removed(mind)
 		ticker.mode.cult -= mind
 
-	species = null
+	if(species)
+		qdel(species)
+		species = null
```

Comment:
Also @unid15 I'm not sure how you mean? This is destroy(), I'm also having the species clear out all the organs.